Use presentation compiler when scaladocing.

vigdorchik opened the pull request:
Don't run unneeded phases when generating scaladoc.
Review by @VladUreche and @dragos.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/745/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1455/

VladUreche commented:
LGTM 
Since there are two reviewers assigned, let's wait for @dragos to comment too.

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1455/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/745/

dragos commented:
:+1: LGTM!

hubertp commented:
Is there any chance to merge forScaladoc and forInteractive?

vigdorchik commented:
On Wed, Oct 17, 2012 at 1:48 PM, Hubert Plociniczak <
notifications@github.com> wrote:

> Is there any chance to merge forScaladoc and forInteractive?
> 
> forScaladoc implies forInteractive, but I currently don't know if the
> reverse is true
> 
>  —
> Reply to this email directly or view it on GitHubhttps://github.com/scala/scala/pull/1503#issuecomment-9521677.

hubertp commented:
Ok. In that case you should also remove !forScaladoc when adapting ConstantType in Typers.scala.

VladUreche commented:
I don't think `forScaladoc` implies `forInteractive` -- running `./scaladoc X.scala`we have `forScaladoc=true` and `forInteractive=false`.

hubertp commented:
Ok, but with this change when running interactive compiler you have forScaladoc and forInteractive both true (so actually my previous comment was wrong, you should remove !forInteractive).

hubertp commented on src/compiler/scala/tools/nsc/doc/DocFactory.scala:
```diff
@@ -32,25 +32,23 @@ import DocParser.Parsed
-      phasesSet += pickler
-      phasesSet += refChecks
+  def mkCompiler = 
+    new Global(settings, reporter) with interactive.RangePositions {
```
the title of the commit is misleading. For me it suggests that we should have new interactive.Global(....)..... or am I missing something?

vigdorchik commented:
@VladUreche right, I mean forInteractive implies forScaladoc.

vigdorchik commented:
@hubertp Would "Use presentation compiler when scaladocing.in the IDE" be clearer?

hubertp commented:
ah, now I get it. You guys probably talked about it so I am missing the context.
Can I know if there is any particular reason why you force scaladoc mode for _every_ usage of interactive.Global? In practice every IDE (not only Eclipse Scala IDE) extends interactive.Global and customizes it by overriding members so it is not obvious for me why we change the behaviour globally.

vigdorchik commented:
On Wed, Oct 17, 2012 at 6:02 PM, Hubert Plociniczak <
notifications@github.com> wrote:

> ah, now I get it. You guys probably talked about it so I am missing the
> context.
> Can I know if there is any particular reason why you force scaladoc mode
> for _every_ usage of interactive.Global? In practice every IDE (not only
> Eclipse Scala IDE) extends interactive.Global and customizes it by
> overriding members so it is not obvious for me why we change the behaviour
> globally.
> 
> Right, didn't think about poor other IDEs :) The idea for eclipse is to
> always keep comments, so that when the IDE asks for them, there is no need
> to reparse. In theory this overriding can be conditional on  some flag set
> dynamically, but that also means another reload and typecheck for
> presenting scaladoc. Do you see many minuses in always keeping doc comments
> in all IDEs?
> 
>  —
> Reply to this email directly or view it on GitHubhttps://github.com/scala/scala/pull/1503#issuecomment-9528247.

hubertp commented:
Personally, I was just interested why you need to have it. There are things like Repl in interactive package which probably don't really want to typecheck docs (not that anyone would write it but still). I'm just not a fun of setting some values implicitly and globally.

Finally we have onlyPresentation def in nsc.Global which basically combined forScaladoc and forInteractive so I am being picky here to make sure that what you change here is not a mistake.

vigdorchik commented:
On Wed, Oct 17, 2012 at 6:36 PM, Hubert Plociniczak <
notifications@github.com> wrote:

> Personally, I was just interested why you need to have it. There are
> things like Repl in interactive package which probably don't really want to
> typecheck docs (not that anyone would write it but still). I'm just not a
> fun of setting some values implicitly and globally.
> 
> Ok then. How about giving a parameter to interactive.Global that defaults
> to false?
> 
> Finally we have onlyPresentation def in nsc.Global which basically
> combined forScaladoc and forInteractive so I am being picky here to make
> sure that what you change here is not a mistake.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/scala/scala/pull/1503#issuecomment-9529544.

vigdorchik committed:
Use presentation compiler when producing documentation from the IDE.
Remove unneeded phases from the scaladoc's own compiler.
Review by @VladUreche.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/750/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1460/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1460/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/750/

hubertp commented:
Thanks, that looks better, I think.

VladUreche commented:
LGTM for the latest commit.
